### ISSUE TITLE
Frigate Controller

### ISSUE THREAD

**Issue by 5olu-s** [LemADEC/WarpDrive#378]
I set up the frigate core with dimensions that are within its parameters but I am unable to make the ship move. I enter in the minimum blocks I can move and the ship doesn't do anything. No text pops up in the chat saying that something is wrong like it usually does and there is no red text in the computer terminal either. The shuttle ship core works like a charm though.

This is on a server as well I should mention.

[2019-06-04-1.log](https://github.com/LemADEC/WarpDrive/files/3261292/2019-06-04-1.log)

**Comment by LemADEC** [comment]
No jump was started because the ship fails the mass requirements:
```
[07:31:19] [Server thread/INFO] [warpdrive]: TileEntityShipCore 'Destiny' @ solarSystem (-38467 211 31879) Connected messageToAllPlayersInArea: Movement aborted!
Ship is too small, min is 10368 block.\nDowngrade ship core or increase mass.
...
[07:33:20] [Server thread/INFO] [warpdrive]: TileEntityShipCore 'Destiny' @ world (1426 116 1926) Connected messageToAllPlayersInArea: Ship is too big for a planet.\nReduce your mass below 3000 blocks.
[07:33:31] [Server thread/WARN] [net.minecraft.server.MinecraftServer]: Can't keep up! Did the system time change, or is the server overloaded? Running 3647ms behind, skipping 72 tick(s)
```
There is a big red warning when you start the computer and at the bottom of the screen.
Players only get notified when dimension are set.

**Comment by 5olu-s** [comment]
That most likely was an older message. I was able to meet the mass requirements in time as I kept adding more and more blocks until the red text disappeared. The interface said it was a viable ship and when I tried to move I was not able to. It gave me no chat indication or any indication as to why it would not start the jump. 

The log attached now is the most recent one I have tried with only WarpDrive and CC tweaks installed.
I had put both the core and the computer fresh in this round.

I bet nothing will be found as nothing happened. I tried jumping three times and attempted hyperspace once.

[latest.log](https://github.com/LemADEC/WarpDrive/files/3263312/latest.log)

**Comment by LemADEC** [comment]
Enable the following logs and try again: 
enable_LUA_logs
enable_break_place_logs
enable_building_logs
enable_chunk_loading_logs
enable_collection_logs
enable_jump_logs
enable_jumpblocks_logs
enable_profiling_CPU_time
enable_profiling_thread_safety
enable_radar_logs
enable_starmap_logs


We need the debug.log file. The latest.log file is incomplete.

**Comment by 5olu-s** [comment]
Alright I enabled all of the settings and here is the debug.log file.


[debug.log](https://github.com/LemADEC/WarpDrive/files/3263423/debug.log)

**Comment by LemADEC** [comment]
```
[16:44:26] [ComputerCraft-Coroutine-0/INFO] [warpdrive]: [CC] LUA call @ solarSystem (-38467 211 31879) from 18:left to warpdriveShipCore.dim_positive(80.0, 46.0, 10.0)
[16:44:26] [ComputerCraft-Coroutine-0/INFO] [warpdrive]: [CC] LUA call is returning 80, 46, 10
[16:44:26] [ComputerCraft-Coroutine-0/INFO] [warpdrive]: [CC] LUA call @ solarSystem (-38467 211 31879) from 18:left to warpdriveShipCore.dim_negative(15.0, 46.0, 15.0)
[16:44:26] [ComputerCraft-Coroutine-0/INFO] [warpdrive]: [CC] LUA call is returning 15, 46, 15
```
=> (80 + 1 +15) * (46 + 1 + 46) * (10 + 1 +15) = 229632
A- Your ship bounding box spans over 200k blocks, that's huge, you better be sure it's not overestimated.

```
[16:44:26] @ solarSystem (-38467 211 31879) Connected was resized, updating...
...
[16:44:37] @ solarSystem (-38467 211 31879) Connected scanning done: mass 10502, volume 26453
```
With your server setup it takes ~11 seconds to scan your ship bounding box.

```
[16:44:43] @ solarSystem (-38467 211 31879) from 18:left to warpdriveShipCore.command("MANUAL", true)
[16:44:43] @ solarSystem (-38467 211 31879) Connected Command set to MANUAL (true)
...
[16:44:43] @ solarSystem (-38467 211 31879) Connected state ONLINE -> initiating jump
[16:44:43] @ solarSystem (-38467 211 31879) Connected scanning started
```
=> new jump setup started, engaging a new scan
B- the scan shouldn't have started, it's configuration bug

```
[16:44:49] @ solarSystem (-38467 211 31879) from 18:left to warpdriveShipCore.command("MANUAL", false)
[16:44:49] @ solarSystem (-38467 211 31879) Connected Command set to MANUAL (false)
```
=> new jump setup started, cancelling the current one while shape was being scanned
```
[16:44:49] @ solarSystem (-38467 211 31879) from 18:left to warpdriveShipCore.command("MANUAL", true)
[16:44:49] @ solarSystem (-38467 211 31879) Connected Command set to MANUAL (true)
```
=> new jump has been confirmed
```
[16:44:54] @ solarSystem (-38467 211 31879) Connected scanning done: mass 10502, volume 26453
[16:44:54] @ solarSystem (-38467 211 31879) Connected state ONLINE -> initiating jump
[16:44:54] @ solarSystem (-38467 211 31879) Connected scanning started
```
=> previous scan is finished, jump request is processed, triggering a new scan...
```
[16:44:55] @ solarSystem (-38467 211 31879) from 18:left to warpdriveShipCore.command("MANUAL", false)
[16:44:55] @ solarSystem (-38467 211 31879) Connected Command set to MANUAL (false)
```
=> new jump setup started, cancelling the current one while shape was being scanned
```
[16:45:00] @ solarSystem (-38467 211 31879) from 18:left to warpdriveShipCore.command("HYPERDRIVE", true)
[16:45:00] @ solarSystem (-38467 211 31879) Connected Command set to HYPERDRIVE (true)
```
=> hyperdrive is confirmed, we're still waiting for previous shape scanning
```
[16:45:06] @ solarSystem (-38467 211 31879) Connected scanning done: mass 10502, volume 26453
[16:45:06] @ solarSystem (-38467 211 31879) Connected state ONLINE -> initiating jump
[16:45:06] @ solarSystem (-38467 211 31879) Connected scanning started
```
=> shape scanning is done, processing hyperjump request, triggering new scanning
```
[16:45:18] @ solarSystem (-38467 211 31879) Connected scanning done: mass 10502, volume 26453
[16:45:18] @ solarSystem (-38467 211 31879) Connected state ONLINE -> initiating jump
[16:45:18] @ solarSystem (-38467 211 31879) Connected scanning started
```
=> shape scanning is done, processing hyperjump request, triggering new scanning.
```
[16:45:24] [Server thread/INFO] [net.minecraft.server.integrated.IntegratedServer]: Saving and pausing game...
```
=> closing game while scan is still ongoing

Long story short:
A- your ship dimensions are insane, make sure they're as small as possible.
B- in the configuration file, change volume_scan_age_tolerance is actually measured in ticks due to a bug in jump sequencer. Change its value from 120 to 2400 until it gets fixed.
C- we probably need a guide message when scan is ongoing during a jump request.

**Comment by LemADEC** [comment]
Remember to set back your log levels to default, at least the enable_jumpblocks_logs one which is very spammy.

**Comment by 5olu-s** [comment]
Alright will do
